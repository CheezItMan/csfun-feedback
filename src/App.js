import './App.css';

import React, { Suspense } from 'react';
import firebaseWrapper from './context/auth'
import StudentsProvider from './providers/StudentsProvider';

// Only load the components being used
const AuthenticatedClassroomApp = React.lazy(() => import('./components/App/AuthenticatedClassroomApp'));
const UnAuthenticatedClassroomApp = React.lazy(() => import('./components/App/UnAuthenticatedClassroomApp'));

const App = (props) => {
  const { user, signOut, signInWithGoogle, signInWithGithub } = props;
  if (user) {
    return (
      <Suspense fallback={<div>Loading</div>} >
        <StudentsProvider>
          <AuthenticatedClassroomApp user={user} signOut={signOut} fallback={<div>Loading</div>} />
        </StudentsProvider>
      </Suspense>
    );
  }
  return (
    <main>
      <h1>Please Log in</h1>
      <Suspense fallback={<div>Loading</div>} >
        <UnAuthenticatedClassroomApp signInWithGoogle={signInWithGoogle} signInWithGithub={signInWithGithub} fallback={<div>Loading</div>} />
      </Suspense>
    </main>
  )
};

export default firebaseWrapper(App);
