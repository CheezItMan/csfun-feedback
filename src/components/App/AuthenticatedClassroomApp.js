import React from 'react';


const AuthenticatedClassroomApp = (props) => {
  return (
    <div>
      <h2>Welcome {props.user.displayName} </h2>

      <button className="btn btn-primary" onClick={() => props.signOut()}>
        Logout
      </button>
    </div>
  );
}

export default AuthenticatedClassroomApp;