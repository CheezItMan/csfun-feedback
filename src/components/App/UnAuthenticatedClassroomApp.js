import React from 'react';


const UnAuthenticatedClassroomApp = (props) => {
  return (
    <div>
      <h2>UnAuthenticated</h2>
      <button onClick={props.signInWithGoogle}>Sign in with Google</button>
      <button onClick={props.signInWithGithub}>Sign in with Github</button>

    </div>
  );
}

export default UnAuthenticatedClassroomApp;