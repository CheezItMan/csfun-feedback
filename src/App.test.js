import React from 'react';
import { render, fireEvent } from 'react-testing-library'
import App from './App';
import { exportAllDeclaration } from '@babel/types';

it('renders without crashing', () => {
  const signInWithGithub = jest.fn();
  const signOut = jest.fn();
  const user = {};


  const div = document.createElement('div');
  const container = render(<App
    signInWithGithub={signInWithGithub}
    user={user}
    signOut={signOut}
  />, div);

  expect(container.firstChild).toMatchSnapshot();

});
