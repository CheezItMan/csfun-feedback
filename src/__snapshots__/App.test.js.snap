// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<div>
  <h2>
    Welcome 
     
  </h2>
  <table
    class="Student-list"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Cohort
        </th>
        <th>
          Class
        </th>
        <th>
          Github Name
        </th>
        <th>
          Email
        </th>
        <th>
           
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
  <button
    class="btn btn-primary"
  >
    Logout
  </button>
</div>
`;

exports[`App will redirect the user to login screen when logged out 1`] = `
<main>
  <h1>
    Please Log in
  </h1>
  <div>
    <h2>
      UnAuthenticated
    </h2>
    <button>
      Sign in with Github
    </button>
  </div>
</main>
`;

exports[`App will redirect the user to show AuthenticatedApp when logged in 1`] = `
<div>
  <h2>
    Welcome 
    Alice Cooper
     
  </h2>
  <table
    class="Student-list"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Cohort
        </th>
        <th>
          Class
        </th>
        <th>
          Github Name
        </th>
        <th>
          Email
        </th>
        <th>
           
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
  <button
    class="btn btn-primary"
  >
    Logout
  </button>
</div>
`;
